submarine-server/server-rpc/src/main/java/org/apache/submarine/server/rpc/SubmarineRpcServer.java

```diff
@@ -0,0 +1,407 @@
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getInput())) {
+      runJobParameters.setInputPath(runJobParameterProto.getInput());
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getKeytab())) {
+      runJobParameters.setKeytab(runJobParameterProto.getKeytab());
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getName())) {
+      runJobParameters.setName(runJobParameterProto.getName());
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getPrincipal())) {
+      runJobParameters.setPrincipal(runJobParameterProto.getPrincipal());
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getQueue())) {
+      runJobParameters.setQueue(runJobParameterProto.getQueue());
+    }
+    if (StringUtils.isNotBlank(runJobParameterProto.getSavedModelPath())) {
+      runJobParameters.setSavedModelPath(
+          runJobParameterProto.getSavedModelPath());
+    }
+
+    runJobParameters.setConfPairs(runJobParameterProto.getConfPairsList())
+        .setDistributed(runJobParameterProto.getDistributed())
+        .setDistributeKeytab(runJobParameterProto.getDistributeKeytab())
+        .setLocalizations(convertLocalizationProtoToLocalization(
+            runJobParameterProto.getLocalizationsList()))
+        .setQuicklinks(convertQuicklinkProtoToQuicklink(
+            runJobParameterProto.getQuicklinksList()))
+        .setSecurityDisabled(runJobParameterProto.getSecurityDisabled())
+        .setWaitJobFinish(runJobParameterProto.getWaitJobFinish())
+        .setWorkerParameter(convertRoleParameterProtoToRoleParameters(
+            runJobParameterProto.getWorkerParameter(), framework));
+    runJobParameters.setEnvars(runJobParameterProto.getEnvarsList());
+
+    ParametersHolder parameter = ParametersHolder.create();
+    parameter.setParameters(runJobParameters);
+    parameter.setFramework(framework);
+    return parameter;
+  }
+
+  private static RoleParameters convertRoleParameterProtoToRoleParameters(
+      RoleParameterProto roleParameterProto, Framework framework) {
+    Role role = null;
+    switch (framework) {
+      case TENSORFLOW:
+        role = TensorFlowRole.valueOf(roleParameterProto.getRole());
+        break;
+      case PYTORCH:
+        role = PyTorchRole.valueOf(roleParameterProto.getRole());
+        break;
+    }
+
+    RoleParameters roleParameters = RoleParameters.createEmpty(role);
+    if (StringUtils.isNotBlank(roleParameterProto.getDockerImage())) {
+      roleParameters.setDockerImage(roleParameterProto.getDockerImage());
+    }
+    if (StringUtils.isNotBlank(roleParameterProto.getLaunchCommand())) {
+      roleParameters.setLaunchCommand(roleParameterProto.getLaunchCommand());
+    }
+
+    roleParameters.setReplicas(roleParameterProto.getReplicas())
+        .setResource(convertResourceProtoToResource(
+            roleParameterProto.getResourceProto()));
+    return roleParameters;
+  }
+
+  private static List<Localization> convertLocalizationProtoToLocalization(
```

Comment:
I'm thinking if the localization parameter with a "local:///home/user/foo/test.py:./test.py" would work in RPC server side. It seems the first local path can only be remote url like hdfs or S3.etc.

Reply:
Yeah, The localization like "local:///home/user/foo/test.py:./test.py" would not work.